src/main/java/org/datadog/jenkins/plugins/datadog/clients/ConcurrentMetricCounters.java

```diff
@@ -2,13 +2,31 @@
 
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.locks.ReentrantLock;
 
 public class ConcurrentMetricCounters {
-    private static final ConcurrentMap<CounterMetric, Integer> metrics = new ConcurrentHashMap<>();
-    public static final ThreadLocal<ConcurrentMap<CounterMetric, Integer>> Counters =
-            new ThreadLocal<ConcurrentMap<CounterMetric, Integer>>() {
-                @Override protected ConcurrentMap<CounterMetric, Integer> initialValue() {
-                    return metrics;
+    private static ConcurrentMetricCounters instance;
+    private static ConcurrentMap<CounterMetric, Integer> metrics = new ConcurrentHashMap<>();
+    public static ReentrantLock lock = new ReentrantLock();
+
+    private ConcurrentMetricCounters(){}
+
+    public static ConcurrentMetricCounters getInstance(){
+        if(instance == null){
+            synchronized (ConcurrentMetricCounters.class) {
```

Comment:
Shouldn't we make the entire method `synchronized` ?

Reply:
see my comment above